lib/elixir/test/elixir/calendar_test.exs

```diff
@@ -624,6 +624,195 @@ defmodule DateTimeTest do
              }
   end
 
+  test "from_iso8601/1 with tz offsets that carry over date boundry" do
+    # backwards past midnight
+    assert DateTime.from_iso8601("2017-06-02T02:30:15+04:00")
```

Comment:
Minor nitpick: what do you think about using this pattern 
```elixir
assert DateTime.from_iso8601("...") == {:ok, %DateTime{...}}
``` 
to remove the pipe `|>` and the `elem(1)`?

Reply:
I'd be okay with that. I was aiming to be consistent with the style of the other tests by copying from here:

https://github.com/elixir-lang/elixir/blob/2d4be9850b04ded001d73464c66c04ffcd00bcce/lib/elixir/test/elixir/calendar_test.exs#L529-L532